# Dia 01 - importar os dados
Primeiro vamos intalar e importar os pandas


In [2]:
pip install pandas

In [3]:
import pandas as pd

---

Vamos agora importar as tabelas

In [4]:
dados_2010_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20101.csv?raw=true')
dados_2010_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20102.csv?raw=true')
dados_2011_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20111.csv?raw=true')
dados_2011_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20112.csv?raw=true')
dados_2012_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20121.csv?raw=true')
dados_2012_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20122.csv?raw=true')
dados_2013_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20131.csv?raw=true')
dados_2013_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20132.csv?raw=true')
dados_2014_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20141.csv?raw=true')
dados_2014_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20142.csv?raw=true')
dados_2015_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20151.csv?raw=true')
dados_2015_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20152.csv?raw=true')
dados_2016_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20161.csv?raw=true')
dados_2016_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20162.csv?raw=true')
dados_2017_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20171.csv?raw=true')
dados_2017_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20172.csv?raw=true')
dados_2018_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20181.csv?raw=true')
dados_2018_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20182.csv?raw=true')
dados_2019_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20191.csv?raw=true')
dados_2019_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20192.csv?raw=true')
dados_2020_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20201.csv?raw=true')


Vamos concatenar os dados

In [10]:
emprestimos = pd.concat([dados_2010_1, dados_2010_2, dados_2011_1, dados_2011_2, dados_2012_1, dados_2012_2, dados_2013_1, dados_2013_2, dados_2014_1, dados_2014_2, dados_2015_1, dados_2015_2, dados_2016_1, dados_2016_2, dados_2017_1,dados_2017_2,dados_2018_1, dados_2018_2, dados_2019_1, dados_2019_2, dados_2020_1], ignore_index = True)
display(emprestimos)

id_emprestimo codigo_barras                 data_renovacao  \
0               709684       L095049                            NaN   
1               709685       L167050                            NaN   
2               709686    2006017618  2010/01/26 08:07:01.738000000   
3               709687       L184117  2010/01/18 11:07:46.470000000   
4               709684       L095049                            NaN   
...                ...           ...                            ...   
2258013        2989086    2008047044                            NaN   
2258014        2989087    2008047047                            NaN   
2258015        2989088    2010052905                            NaN   
2258016        2989089    2010048023                            NaN   
2258017        2989090    2013106838                            NaN   

                       data_emprestimo                 data_devolucao  \
0        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
1        2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
2        2010/01/04 08:08:44.081000000  2010/02/25 07:36:25.800000000   
3        2010/01/04 08:24:21.284000000  2010/02/03 08:58:45.692000000   
4        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
...                                ...                            ...   
2258013  2020/05/05 19:51:06.059000000  2021/05/13 14:53:31.000000000   
2258014  2020/05/05 19:51:06.135000000  2021/05/13 14:53:38.000000000   
2258015  2020/05/05 19:51:06.164000000  2021/05/13 15:01:46.000000000   
2258016  2020/06/09 11:54:02.669000000  2021/07/21 15:33:55.000000000   
2258017  2020/06/09 11:54:02.852000000  2021/07/21 15:33:45.000000000   

         matricula_ou_siape             tipo_vinculo_usuario  
0              2.008023e+09               ALUNO DE GRADUAÇÃO  
1              2.008023e+09               ALUNO DE GRADUAÇÃO  
2              2.008112e+09           ALUNO DE PÓS-GRADUAÇÃO  
3              2.007211e+08               ALUNO DE GRADUAÇÃO  
4              2.008023e+09               ALUNO DE GRADUAÇÃO  
...                     ...                              ...  
2258013        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
2258014        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
2258015        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
2258016        2.020101e+10           ALUNO DE PÓS-GRADUAÇÃO  
2258017        2.020101e+10           ALUNO DE PÓS-GRADUAÇÃO  

[2258018 rows x 7 columns]

Verificando a quantidade de linhas

In [11]:
emprestimos.value_counts()

id_emprestimo  codigo_barras  data_renovacao                 data_emprestimo                data_devolucao                 matricula_ou_siape  tipo_vinculo_usuario  
709705         2009032168     2010/01/17 18:56:43.909000000  2010/01/04 09:58:46.549000000  2010/01/20 13:14:34.464000000  2.008010e+09        ALUNO DE GRADUAÇÃO        7
709704         2006021442     2010/01/17 18:56:43.923000000  2010/01/04 09:58:46.522000000  2010/02/01 13:59:02.528000000  2.008010e+09        ALUNO DE GRADUAÇÃO        7
709698         2009047725     2010/01/18 14:44:41.163000000  2010/01/04 09:21:19.099000000  2010/02/02 12:02:38.444000000  2.009047e+09        ALUNO DE GRADUAÇÃO        3
2351375        L057436        2016/01/31 07:30:36.293000000  2016/01/02 08:38:40.009000000  2016/02/21 13:16:29.000000000  2.016101e+10        ALUNO DE PÓS-GRADUAÇÃO    2
2351376        H004983        2016/01/31 07:31:36.786000000  2016/01/02 08:46:42.991000000  2016/02/22 16:10:28.000000000  2.016101e+10        ALUNO DE PÓS-GRADUAÇÃO    2
                                                                                                                                                                        ..
1443393        2012037906     2013/03/20 20:17:10.574000000  2013/03/07 11:24:19.326000000  2013/03/25 08:58:44.000000000  2.011038e+09        ALUNO DE GRADUAÇÃO        1
1443395        2011033069     2013/04/07 01:06:51.679000000  2013/03/07 11:25:22.661000000  2013/05/07 16:19:34.000000000  2.011106e+09        ALUNO DE PÓS-GRADUAÇÃO    1
1443401        2011078773     2013/03/22 18:04:25.360000000  2013/03/07 11:26:28.326000000  2013/04/03 16:53:35.000000000  2.012038e+09        ALUNO DE GRADUAÇÃO        1
1443402        2012043035     2013/03/14 09:32:24.614000000  2013/03/07 11:27:24.876000000  2013/04/01 08:58:02.000000000  2.013025e+09        ALUNO DE GRADUAÇÃO        1
1443366        2010049712     2013/03/20 20:11:43.832000000  2013/03/07 11:10:58.775000000  2013/04/04 08:27:18.000000000  2.012936e+09        ALUNO DE GRADUAÇÃO        1
Name: count, Length: 968028, dtype: int64

 Vemos que temos 968028 linhas , <br>
 Vamos excluir as linhas duplicadas:

In [12]:
emprestimos.drop_duplicates()
display(emprestimos)

id_emprestimo codigo_barras                 data_renovacao  \
0               709684       L095049                            NaN   
1               709685       L167050                            NaN   
2               709686    2006017618  2010/01/26 08:07:01.738000000   
3               709687       L184117  2010/01/18 11:07:46.470000000   
4               709684       L095049                            NaN   
...                ...           ...                            ...   
2258013        2989086    2008047044                            NaN   
2258014        2989087    2008047047                            NaN   
2258015        2989088    2010052905                            NaN   
2258016        2989089    2010048023                            NaN   
2258017        2989090    2013106838                            NaN   

                       data_emprestimo                 data_devolucao  \
0        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
1        2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
2        2010/01/04 08:08:44.081000000  2010/02/25 07:36:25.800000000   
3        2010/01/04 08:24:21.284000000  2010/02/03 08:58:45.692000000   
4        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
...                                ...                            ...   
2258013  2020/05/05 19:51:06.059000000  2021/05/13 14:53:31.000000000   
2258014  2020/05/05 19:51:06.135000000  2021/05/13 14:53:38.000000000   
2258015  2020/05/05 19:51:06.164000000  2021/05/13 15:01:46.000000000   
2258016  2020/06/09 11:54:02.669000000  2021/07/21 15:33:55.000000000   
2258017  2020/06/09 11:54:02.852000000  2021/07/21 15:33:45.000000000   

         matricula_ou_siape             tipo_vinculo_usuario  
0              2.008023e+09               ALUNO DE GRADUAÇÃO  
1              2.008023e+09               ALUNO DE GRADUAÇÃO  
2              2.008112e+09           ALUNO DE PÓS-GRADUAÇÃO  
3              2.007211e+08               ALUNO DE GRADUAÇÃO  
4              2.008023e+09               ALUNO DE GRADUAÇÃO  
...                     ...                              ...  
2258013        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
2258014        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
2258015        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
2258016        2.020101e+10           ALUNO DE PÓS-GRADUAÇÃO  
2258017        2.020101e+10           ALUNO DE PÓS-GRADUAÇÃO  

[2258018 rows x 7 columns]

Conseguimos ver que a quantidade foi diminuida para 2258018